### PR TITLE
Jetty 12.0.x use Maven build cache for PRs and branches except jetty-12.0.x main branch

### PR DESCRIPTION
**Pull request by olamy**
- apply configuration/changes for build cache
- fix Jenkinsfile
- do not run javadoc for now
- fix paste
- fix paste
- manage properly the need for a jetty-util classes directory
- do not attach zip output for remote, lazyRestore true and restoreGeneratedSources false
- only apache snapshots
- disable cache during release
- disable cache during dependencies report update
- test javadoc as part of ci profile
- add logic to not build cache when main branch or label no-build-cache
- fix logic

### CODE REVIEW

**Comment by lorban** [comment]
Could you please move the EE9 (demos, maven plugin...) changes to another PR?

I would very much prefer to see build-only changes in this PR.

File: documentation/jetty-documentation/src/main/asciidoc/contribution-guide/4-build.adoc
Comment (lorban) [inline documentation/jetty-documentation/src/main/asciidoc/contribution-guide/4-build.adoc]: HTTP/3 works fine with JDK 11, 17 is only needed if you want to use the Foreign API binding instead of the JNA one.

File: documentation/jetty-documentation/src/main/asciidoc/contribution-guide/4-build.adoc
Comment (lorban) [inline documentation/jetty-documentation/src/main/asciidoc/contribution-guide/4-build.adoc]: HTTP/3 works fine with JDK 11, 17 is only needed if you want to use the Foreign API binding instead of the JNA one.

File: documentation/jetty-documentation/src/main/asciidoc/contribution-guide/4-build.adoc
Comment (lorban) [inline documentation/jetty-documentation/src/main/asciidoc/contribution-guide/4-build.adoc]: We do not support (yet) the JDK 19 Foreign API, so 17 still is your only option if you want to use the Foreign API binding.

**Comment by olamy** [comment]
> Could you please move the EE9 (demos, maven plugin...) changes to another PR?
> 
> I would very much prefer to see build-only changes in this PR.

No because those changes are needed to have the build cache working properly
There are few fixes similar to this issue (https://github.com/eclipse/jetty.project/issues/10312)
some demos etc rely on hardcoded path to another module build output and without declaring this dependency which is wrong (but most of the time not visible the use of build cache/incremental just fail because of a wrong build)

File: documentation/jetty-documentation/src/main/asciidoc/contribution-guide/4-build.adoc
Comment (olamy) [reply documentation/jetty-documentation/src/main/asciidoc/contribution-guide/4-build.adoc]: ah yes I can change that

**Comment by lorban** [comment]
> > Could you please move the EE9 (demos, maven plugin...) changes to another PR?
> > I would very much prefer to see build-only changes in this PR.
> 
> No because those changes are needed to have the build cache working properly There are few fixes similar to this issue (#10312) some demos etc rely on hardcoded path to another module build output and without declaring this dependency which is wrong (but most of the time not visible the use of build cache/incremental just fail because of a wrong build)

If the build cache changes depend on the EE9 changes, couldn't the latter be merged first, in an independent PR?

**Comment by olamy** [comment]
Not sure to understand your comment?
As I said in my previous comment the cache will not work without those changes as technically those changes fix the build which by chance was not broken.
So those changes are needed otherwise the build cache may not work properly

File: documentation/jetty-documentation/src/main/asciidoc/contribution-guide/4-build.adoc
Comment (olamy) [reply documentation/jetty-documentation/src/main/asciidoc/contribution-guide/4-build.adoc]: removed

File: documentation/jetty-documentation/src/main/asciidoc/contribution-guide/4-build.adoc
Comment (olamy) [reply documentation/jetty-documentation/src/main/asciidoc/contribution-guide/4-build.adoc]: removed

File: documentation/jetty-documentation/src/main/asciidoc/contribution-guide/4-build.adoc
Comment (olamy) [reply documentation/jetty-documentation/src/main/asciidoc/contribution-guide/4-build.adoc]: removed

**Comment by gregw** [comment]
@olamy what is the status of t his. Faster CI builds would be very much appreciated.  Is it just waiting for review or does it need more work?

**Comment by olamy** [comment]
> @olamy what is the status of t his. Faster CI builds would be very much appreciated. Is it just waiting for review or does it need more work?

@gregw  just waiting for review and approval  otherwise I cannot merge it :) 
if it causes a problem it's very easy to disable. 
just have a try with the branch locally

File: .mvn/jvm.config
Comment (gregw) [inline .mvn/jvm.config]: no line termination

File: .mvn/maven-build-cache-config.xml
Comment (gregw) [inline .mvn/maven-build-cache-config.xml]: no line termination

File: .mvn/maven.config
Comment (gregw) [inline .mvn/maven.config]: no line termination

File: jetty-ee10/jetty-ee10-demos/jetty-ee10-demo-embedded/src/main/java/org/eclipse/jetty/ee10/demos/JettyDemos.java
Comment (gregw) [inline jetty-ee10/jetty-ee10-demos/jetty-ee10-demo-embedded/src/main/java/org/eclipse/jetty/ee10/demos/JettyDemos.java]: no line termination

File: jetty-ee9/jetty-ee9-demos/jetty-ee9-demo-embedded/src/main/java/org/eclipse/jetty/ee9/demos/JettyDemos.java
Comment (gregw) [inline jetty-ee9/jetty-ee9-demos/jetty-ee9-demo-embedded/src/main/java/org/eclipse/jetty/ee9/demos/JettyDemos.java]: no line termination

**Review by gregw — DISMISSED** [review]
I could of niggles.... plus I'm not really qualified to deeply review maven magic.
But I can't see anything obviously wrong... and if it builds quickly and is able to be disabled if need be!!!

File: jetty-ee10/jetty-ee10-demos/jetty-ee10-demo-embedded/src/main/java/org/eclipse/jetty/ee10/demos/JettyDemos.java
Comment (gregw) [reply jetty-ee10/jetty-ee10-demos/jetty-ee10-demo-embedded/src/main/java/org/eclipse/jetty/ee10/demos/JettyDemos.java]: still missing

**Review by gregw — APPROVED** [review]

**Pull request merged**
